src/server/server_state.cc

```diff
@@ -181,4 +181,10 @@ void ServerState::ReturnInterpreter(Interpreter* ir) {
   interpreter_mgr_.Return(ir);
 }
 
+ServerState* ServerState::SafeTLocal() {
+  // https://stackoverflow.com/a/75622732
```

Comment:
no need for `asm volatile("");` imho.
on the other hand, I am not sure if actually does anything in this one-liner.

Reply:
> noinline
> This function attribute prevents a function from being considered for inlining. If the function does not have side effects, there are optimizations other than inlining that cause function calls to be optimized away, although the function call is live. To keep such calls from being optimized away, put asm ...

https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#index-noinline-function-attribute

I assume that _optimizations other than inlining_ mean reordering and / or even total dead code elimination, because I can't imagine anything that'll break it. But it won't hurt keeping it because it's scope is like one line